src/python/pants/engine/legacy/address_mapper.py

```diff
@@ -73,17 +73,24 @@ def _internal_scan_specs(self, specs, fail_fast=True, missing_is_fatal=True):
     result = self._engine.execute(request)
     if result.error:
       raise self.BuildFileScanError(str(result.error))
-    root_entries = self._scheduler.root_entries(request)
+    root_entries = result.root_products
```

Comment:
Noticed that the root_entries call here was unnecessary when covering the case where an unexpected Throw value is hit.